onadata/apps/logger/management/commands/transfer_project.py

```diff
@@ -0,0 +1,125 @@
+"""Functionality to transfer a project form one owner to another."""
+
+from django.contrib.auth import get_user_model
+from django.core.management.base import BaseCommand
+from django.db import transaction
+
+from onadata.apps.logger.models import Project, XForm, DataView, MergedXForm
+from onadata.apps.logger.models.project import set_object_permissions \
+    as set_project_permissions
+from onadata.libs.utils.project_utils import set_project_perms_to_xform
+from onadata.libs.utils.viewer_tools import get_form_url, enketo_url
+from onadata.apps.main.models.meta_data import unique_type_for_form
+
+
+class Command(BaseCommand):  # pylint: disable=C0111
+    help = 'A command to reassign a project form one user to the other.'
+
+    errors = []
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--currentowner',
+            help='Username of the current owner of of the projects',
+        )
+        parser.add_argument(
+            '--newowner',
+            help='TUsername of new owner of the projects',
+        )
+        parser.add_argument(
+            '--httphost',
+            help='The http host for the server e.g ona.io',
+        )
+        parser.add_argument(
+            '--httpprotocol',
+            help='The protocol to use for enketo url: https or http',
+        )
+
+    def get_user(self, username):  # pylint: disable=C0111
+        user_model = get_user_model()
+        user = None
+        try:
+            user = user_model.objects.get(username=username)
+        except user_model.DoesNotExist:
+            self.errors.append("User {0} does not exist \n".format(username))
+        return user
+
+    def update_xform_with_new_user(self, project, user):
+        """
+        Update XForm user update the DataViews and also set the permissions
+        for the xform and the project.
+        """
+        xforms = XForm.objects.filter(
+            project=project, deleted_at__isnull=True, downloadable=True)
+        for form in xforms:
+            form.user = user
+            form.created_by = user
+            form.save()
+            self.update_data_views(form)
+            set_project_perms_to_xform(form, project)
+            self.update_enketo_urls(form)
+
+    @staticmethod
+    def update_data_views(form):
+        """Update DataView project for the XForm given. """
+        dataviews = DataView.objects.filter(
+            xform=form, project=form.project, deleted_at__isnull=True)
+        for data_view in dataviews:
+            data_view.project = form.project
+            data_view.save()
+
+    @staticmethod
+    def update_merged_xform(project, user):
+        merged_xforms = MergedXForm.objects.filter(
+            project=project, deleted_at__isnull=True)
+        for form in merged_xforms:
+            form.user = user
+            form.created_by = user
+            form.save()
+            set_project_perms_to_xform(form, project)
+
+    def update_enketo_urls(self, form):
```

Comment:
I think this does correctly create Enketo metadata records for a new Enketo URI from the move but maintains the old URI. I don't think this sufficiently addresses the issue in #845. A fix for that could be having an endpoint that does not include user/org identifying information in the URI.